backend/onyx/document_index/vespa/shared_utils/vespa_request_builders.py

```diff
@@ -106,3 +107,25 @@ def build_vespa_id_based_retrieval_yql(
 
     id_based_retrieval_yql_section += ")"
     return id_based_retrieval_yql_section
+
+
+def build_deletion_selection(doc_id: str, version_cutoff: int, doc_type: str) -> str:
+    """
+    Build a Vespa selection expression that includes:
+      - {doc_type}.document_id == <doc_id>
+      - {doc_type}.doc_updated_at < version_cutoff
+    """
+    # Escape single quotes by doubling them for Vespa selection expressions
+    escaped_doc_id = doc_id.replace("'", "''")
+
+    return (
+        f"({doc_type}.document_id=='{escaped_doc_id}') and "
+        f"({doc_type}.doc_updated_at < {version_cutoff})"
+    )
+
+
+def build_selection_query_param(filter_str: str) -> str:
```

Comment:
Does this get used in more than one place?  Maybe it should just be moved inline with the calling function?